src/hapi/pipelines/database/humanitarian_needs.py

```diff
@@ -38,21 +41,40 @@ def __init__(
         self._configuration = configuration
 
     def get_admin2_ref(self, row, dataset_name, errors):
-        admin_code = row["Admin 2 PCode"]
-        if admin_code == "#adm2+code":  # ignore HXL row
+        countryiso3 = row["Country ISO3"]
+        if countryiso3 == "#country+code":  # ignore HXL row
             return None
-        if admin_code:
-            admin_level = "admintwo"
-        else:
-            admin_code = row["Admin 1 PCode"]
-            if admin_code:
-                admin_level = "adminone"
-            else:
-                admin_code = row["Country ISO3"]
+        admin_level = "0"
+        for header in row:
+            match = self.admin_name_regex.match(header)
+            if match and row[header]:
+                admin_level = match.group(1)
+        match admin_level:
+            case "0":
                 admin_level = "national"
-        return self._admins.get_admin2_ref(
+                admin_code = countryiso3
+            case "1":
+                admin_level = "adminone"
+                admin_code = row["Admin 1 PCode"]
+            case "2":
+                admin_level = "admintwo"
+                admin_code = row["Admin 2 PCode"]
+            case _:
+                return None
+        admin2_ref = self._admins.get_admin2_ref(
             admin_level, admin_code, dataset_name, errors
         )
+        if admin2_ref is None:
+            if admin_level == "adminone":
+                admin_code = f"{countryiso3}-XXX"
```

Comment:
You should use the [method](https://github.com/OCHA-DAP/hapi-pipelines/blob/main/src/hapi/pipelines/database/admins.py#L190)

Reply:
Good idea. I've also made one for admin2 to location.